File: Python3/2033_minimum_operations_to_make_a_uni_value_grid.py
# https://leetcode.cn/problems/minimum-operations-to-make-a-uni-value-grid/

class Solution:
    def minOperations(self, grid: list[list[int]], x: int) -> int:
        mod = grid[0][0] % x
        for row in grid:
            for val in row:
                if val % x != mod:
                    return -1

        res = 0
        rows, cols = len(grid), len(grid[0])
        n = rows * cols
        values = [grid[i//cols][i%cols] for i in range(n)]
        values.sort()
        mid = values[n//2]
        for i in range(n):
            res += abs(values[i] -  mid) // x
        return res

File: Cpp/2033_minimum_operations_to_make_a_uni_value_grid.cpp
// https://leetcode.cn/problems/minimum-operations-to-make-a-uni-value-grid/

#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

class Solution {
public:
    int minOperations(vector<vector<int>>& grid, int x) {
        int res = 0;
        int rows = grid.size(), cols = grid[0].size();
        int n = rows * cols;
        vector<int> values(n);
        
        int mod = grid[0][0] % x;
        for (int i = 0; i < n; ++i) {
            int val = grid[i/cols][i%cols];
            if (val % x != mod) return -1;
            values[i] = val;
        }
        sort(values.begin(), values.end());
        int mid = values[n/2];
        for (const int &val: values) {
            res += abs(val - mid) / x;
        }
        return res;
    }
};

int main() {
    Solution sol;
    vector<vector<int>> grid = {
        {2,4},
        {6,8}
    };
    int x = 2;
    cout << sol.minOperations(grid, x) << '\n';
    return 0;
}